File: ray-tracer/elements/surfaces/Matt.cs
using System;
using ray_tracer.common;
using ray_tracer.scene;


namespace ray_tracer.elements.surfaces
{
    public class Matt : Surface
    {
        public double roughness => 50.0;

        public RColor diffuse(Vector pos)
        {
            return RColor.white;
        }

        public double reflect(Vector pos)
        {
            return 0.05;
        }

        public RColor specular(Vector pos)
        {
            return RColor.grey;
        }
    }
}

File: java-app/src/pl/mikolajczuk/kamil/raytracing/lib/elements/surfaces/Matt.java
package pl.mikolajczuk.kamil.raytracing.lib.elements.surfaces;

import pl.mikolajczuk.kamil.raytracing.lib.common.*;
import pl.mikolajczuk.kamil.raytracing.lib.scene.*;

public class Matt implements Surface {
    public double getRoughness() {
        return 10.0;
    }

    public RColor diffuse(Vector pos) {
        return RColor.white;
    }

    public double reflect(Vector pos) {
        return 0.05;
    }

    public RColor specular(Vector pos) {
        return RColor.grey;
    }
}
